src/Compilers/Core/Portable/DiagnosticAnalyzer/AsyncQueue.cs

```diff
@@ -94,9 +97,12 @@ private bool EnqueueCore(TElement value)
                 waiter = _waiters.Dequeue();
             }
 
-            // Invoke SetResult on a separate task, as this invocation could cause the underlying task to executing,
-            // which could be a long running operation that can potentially cause a deadlock if executed on the current thread.
-            Task.Run(() => waiter.SetResult(value));
+            Debug.Assert(waiter.Task.CreationOptions.HasFlag(TaskCreationOptions.RunContinuationsAsynchronously));
+            if (!waiter.TrySetResult(value))
+            {
+                // A waiter was available in the queue, but was cancelled before we were able to assign this value
```

Comment:
```suggestion
                // A waiter was available in the queue, but was cancelled or ran to completion before we were able to assign this value
```

Reply:
> ... or ran to completion ...

How could this occur?

Reply:
It can't 😄 

More careful reading of the code removed the place where I thought this could happen

Reply:
➡️ No action taken